mobile_packet_verifier/src/daemon.rs

```diff
@@ -92,8 +94,15 @@ where
                 },
                 _ = sleep_until(burn_time) => {
                     // It's time to burn
-                    self.burner.burn(&self.pool).await?;
-                    burn_time = Instant::now() + self.burn_period;
+                    match self.burner.burn(&self.pool).await {
+                        Ok(_) => {
+                            burn_time = Instant::now() + self.burn_period;
+                        }
+                        Err(e) => {
+                            burn_time = Instant::now() + self.min_burn_period;
+                            tracing::warn!("failed to burn {e:?}, re running burn in {:?} min", self.min_burn_period);
+                        }
```

Comment:
I think this is fine and not really needing of a test, but if you want to you can extract the burn time logic into a function and add a unit test